pkgs/desktops/deepin/dde-launcher/default.nix

```diff
@@ -0,0 +1,80 @@
+{ stdenv, fetchFromGitHub, pkgconfig, cmake, dbus,
+  dde-qt-dbus-factory, dde-session-ui, deepin, deepin-desktop-schemas,
+  deepin-wallpapers, dtkcore, dtkwidget, gsettings-qt, qtsvg, qttools,
+  qtx11extras, which, xdg_utils, wrapGAppsHook
+}:
+
+stdenv.mkDerivation rec {
+  name = "${pname}-${version}";
+  pname = "dde-launcher";
+  version = "4.6.11";
+
+  src = fetchFromGitHub {
+    owner = "linuxdeepin";
+    repo = pname;
+    rev = version;
+    sha256 = "00zvnjn6sh9i51l2187jlvpz114pxv12isvji2axnvxaf3kwgc1y";
+  };
+
+  nativeBuildInputs = [
+    cmake
+    pkgconfig
+    qttools
+    wrapGAppsHook
+    deepin.setupHook
+  ];
+
+  buildInputs = [
+    dbus
+    dde-qt-dbus-factory
+    dde-session-ui
+    deepin-desktop-schemas
+    deepin-wallpapers
+    dtkcore
+    dtkwidget
+    gsettings-qt
+    qtsvg
+    qtx11extras
+    which
+    xdg_utils
+  ];
+
+  postPatch = ''
+    # debugging
+    searchHardCodedPaths
+
+    substituteInPlace CMakeLists.txt --replace "/usr/share" "$out/share"
+
+    substituteInPlace src/dbusservices/com.deepin.dde.Launcher.service --replace "/usr" "$out"
+
+    substituteInPlace src/historywidget.cpp --replace "xdg-open" "${xdg_utils}/bin/xdg-open"
+    substituteInPlace src/widgets/miniframebottombar.cpp --replace "dbus-send" "${dbus}/bin/dbus-send"
+    substituteInPlace src/widgets/miniframebottombar.cpp --replace "dde-shutdown" "${dde-session-ui}/bin/dde-shutdown"
+    substituteInPlace src/widgets/miniframenavigation.cpp --replace "dbus-send" "${dbus}/bin/dbus-send"
+    substituteInPlace src/widgets/miniframerightbar.cpp --replace "which" "${which}/bin/which"
+    substituteInPlace src/widgets/searchwidget.cpp --replace "dbus-send" "${dbus}/bin/dbus-send"
```

Comment:
https://github.com/linuxdeepin/dde-launcher/blob/523b1feb0638f7fe23ee6519df8a873b368d8c55/src/widgets/searchwidget.cpp#L81 should return true so this portion of code wouldn't be ran right?

Reply:
`DTK_VERSION >= DTK_VERSION_CHECK(2, 0, 8, 0))`

As dtkcore/dtkwidget current version is 2.0.14, this test is true,  `dbus-send` will not be explicitly run.
Therefore `dbus-send` path does not need to be hard coded, right?
I  am going to remove them from the nix expression.

Reply:
Yep that is what I meant :+1: